09_json/n0npax/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	puppy "github.com/anz-bank/go-course/09_json/n0npax/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/n0npax/pkg/puppy/store"
+)
+
+var (
+	out           io.Writer = os.Stdout
+	args                    = os.Args[1:]
+	puppyFilePath           = kingpin.Flag("data", "path to file with puppies data").Short('d').ExistingFile()
```

Comment:
I think [kingpin.File](https://godoc.org/gopkg.in/alecthomas/kingpin.v2#ArgClause.File) opens the file for you.